### PR TITLE
#3586-UI-Implement positive_subscribe_system

### PR DESCRIPTION
**Pull request by ldjebran**
```console
(2.7) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/ui/test_contentview.py -v -k "test_positive_subscribe_system"
================================================================= test session starts ==================================================================
platform linux2 -- Python 2.7.12, pytest-3.0.3, py-1.4.31, pluggy-0.4.0 -- /home/dlezz/bin/redhat/python/2.7/bin/python
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile: 
plugins: cov-2.4.0, xdist-1.15.0
collected 81 items 

tests/foreman/ui/test_contentview.py::ContentViewTestCase::test_positive_subscribe_system <- robottelo/decorators/__init__.py PASSED

================================================================= 80 tests deselected ==================================================================
====================================================== 1 passed, 80 deselected
```
issue: #3586

### CODE REVIEW

File: tests/foreman/ui/test_contentview.py [hunk 1]
```diff
@@ -2273,11 +2277,113 @@ def test_positive_subscribe_system(self):
...
-
-
         @CaseLevel: Integration
         """
+        cv_name = gen_string('alpha')
+        filter_name = gen_string('alpha')
+        subscription_name = DEFAULT_SUBSCRIPTION_NAME
+        puppet_repo_url = FAKE_0_PUPPET_REPO
+        puppet_module_name = 'httpd'
+        repos = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: basically NACK overall idea to put completely different functionalities into one test case. Divide it into three test cases and remove that complex logic

Comment (ldjebran) [inline tests/foreman/ui/test_contentview.py]: @oshtaier , you mean to make a separate test for each repo?

Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: right, separate test per each type

**Comment by ldjebran** [comment]
made separate test for each repo
```console
dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/ui/test_contentview.py -v -k "test_positive_subscribe_system_with_rh_custom_spin or test_positive_subscribe_system_with_custom_content or test_positive_subscribe_system_with_puppet_modules"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.12, pytest-3.0.5, py-1.4.32, pluggy-0.4.0 -- /home/dlezz/bin/redhat/python/2.7/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile: 
plugins: xdist-1.14, cov-2.3.1
collected 83 items 

tests/foreman/ui/test_contentview.py::ContentViewTestCase::test_positive_subscribe_system_with_custom_content <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_contentview.py::ContentViewTestCase::test_positive_subscribe_system_with_puppet_modules <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_contentview.py::ContentViewTestCase::test_positive_subscribe_system_with_rh_custom_spin <- robottelo/decorators/__init__.py PASSED

================================================= 80 tests deselected ==================================================
====================================== 3 passed, 80 deselected in 1170.76 seconds ======================================
```

**Comment by ldjebran** [comment]
@oshtaier @rochacbruno tests refactored please review

**Comment by oshtaier** [comment]
Could we check that not only operation was successful, but registered host is really present in the system, for example on host page?

**Comment by oshtaier** [comment]
All other changes seems sane and good enough to submit

**Comment by ldjebran** [comment]
will update with a check in content host page that the subscribed host exist and update tests

**Comment by ldjebran** [comment]
```console
platform linux2 -- Python 2.7.12, pytest-3.0.3, py-1.4.31, pluggy-0.4.0 -- /home/dlezz/bin/redhat/python/2.7/bin/python
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile: 
plugins: cov-2.4.0, xdist-1.15.0
collected 83 items 

tests/foreman/ui/test_contentview.py::ContentViewTestCase::test_positive_subscribe_system_with_custom_content <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_contentview.py::ContentViewTestCase::test_positive_subscribe_system_with_puppet_modules <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_contentview.py::ContentViewTestCase::test_positive_subscribe_system_with_rh_custom_spin <- robottelo/decorators/__init__.py PASSED

================================================= 80 tests deselected ==================================================
====================================== 3 passed, 80 deselected in 897.93 seconds =======================================
```
updated tests for last changes with content hosts

File: tests/foreman/ui/test_contentview.py [hunk 2]
```diff
@@ -2253,31 +2256,236 @@ def test_positive_clone_within_diff_env(self):
...
-        # other tests for that.
-        # Variations:
-        # * rh content
-        # * rh custom spins
-        # * custom content
-        # * composite
-        # * CVs with puppet modules
-        """attempt to subscribe systems to content view(s)
+    def test_positive_subscribe_system_with_rh_custom_spin(self):
+        """Attempt to subscribe systems to content view(s)
```
Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: We need to have that string different per each to test case to represent actual actions that we perform

**Comment by oshtaier** [comment]
ACK pending last change. No need to test modifications to docstrings, so once committed, it will be good to merge right away

**Comment by oshtaier** [comment]
ACK

**Pull request merged**